eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/CredentialInfo.java

```diff
@@ -78,16 +78,17 @@ private static CredentialInfo createCredentialInfo(String connectionString, Stri
         if (ImplUtils.isNullOrEmpty(connectionString)) {
             throw new IllegalArgumentException("Connection string is null or empty");
         }
-        CredentialInfo credentialInfo = new CredentialInfo();
-        String[] tokenValuePairs = connectionString.split(TOKEN_VALUE_PAIR_DELIMITER);
+        final CredentialInfo credentialInfo = new CredentialInfo();
+        final String[] tokenValuePairs = connectionString.split(TOKEN_VALUE_PAIR_DELIMITER);
         for (String tokenValuePair : tokenValuePairs) {
-            String[] pair = tokenValuePair.split(TOKEN_VALUE_SEPERATOR);
+            final String[] pair = tokenValuePair.split(TOKEN_VALUE_SEPERATOR, 2);
```

Comment:
@mssfang SharedAccessKeyValue also has a `=` in it, in the previous parsing logic, it was removing the `=`, leaving an invalid token.

Reply:
Resolved in #3859 . This change will go away after the PR is merged.